src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,197 @@
+import React, {
+  useState,
+  useCallback,
+  useEffect,
+  useMemo,
+} from 'react';
+import {
+  getCommentsByPostId,
+  deleteComment,
+} from '../../api/comments';
+import { NewCommentForm } from '../NewCommentForm';
+import { Loader } from '../Loader';
+import { ErrorNotification } from '../ErrorNotification';
+import { Post } from '../../types/Post';
+import { Comment } from '../../types/Comment';
+import { Error } from '../../types/Error';
+
+type Props = {
+  posts: Post[];
+  postId: number | null;
+  error: Error | null;
+  onError: (error: Error | null) => void;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  posts,
+  postId,
+  error,
+  onError,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isCommentForm, setIsCommentForm] = useState(false);
+
+  const handleAddComment = useCallback((newComment) => {
+    setComments(allComments => [...allComments, newComment]);
+  }, []);
+
+  const currentPost
+    = useMemo(() => posts.filter(post => post.id === postId)[0], [postId]);
+
+  const handleDeleteComment = async (commentId: number) => {
+    try {
+      setComments(
+        allComments => allComments.filter(
+          comment => comment.id !== commentId,
+        ),
+      );
+      onError(null);
+
+      if (!commentId) {
+        return;
+      }
+
+      await deleteComment(commentId);
+    } catch {
+      onError(Error.DELETE_COMMENT);
+    }
+  };
+
+  const getCommentList = async (id: number | null) => {
+    try {
+      setIsLoading(true);
+      onError(null);
+      setIsCommentForm(false);
+
+      if (!id) {
+        return;
+      }
+
+      const commentList = await getCommentsByPostId(id);
+
+      setComments(commentList);
+    } catch {
+      onError(Error.GET_COMMENTS);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    if (postId) {
+      getCommentList(postId);
+    }
+  }, [postId]);
+
+  return (
+    <div className="content" data-cy="PostDetails">
+      <div className="content" data-cy="PostDetails">
```

Comment:
The div element with className 'content' is nested within another div with the same className. This is likely an error, please check if this is the intended behavior.